shims/spark311/src/main/scala/com/nvidia/spark/rapids/shims/spark311/ParquetCachedBatchSerializer.scala

```diff
@@ -293,6 +301,7 @@ class ParquetCachedBatchSerializer extends CachedBatchSerializer with Arm {
       case ArrayType(elementType, _) => isTypeSupportedByParquet(elementType)
       case MapType(keyType, valueType, _) => isTypeSupportedByParquet(keyType) &&
           isTypeSupportedByParquet(valueType)
+      case d: DecimalType if d.scale < 0 => false
```

Comment:
Could we add some scaladocs to `isTypeSupportedByParquet` to clarify that this is really types that parquet supports without any type of modification.